token/program-2022/src/processor.rs

```diff
@@ -1371,6 +1371,77 @@ impl Processor {
         Ok(())
     }
 
+    /// Withdraw Excess Lamports is used to recover Lamports transfered to any TokenProgram owned account
+    /// by moving them to another account
+    /// of the source account.
+    pub fn process_withdraw_excess_lamports(
+        program_id: &Pubkey,
+        accounts: &[AccountInfo],
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+
+        let source_info = next_account_info(account_info_iter)?;
+        let destination_info = next_account_info(account_info_iter)?;
+        let authority_info = next_account_info(account_info_iter)?;
+
+        let source_data = source_info.data.borrow();
+
+        if let Ok(account) = StateWithExtensions::<Account>::unpack(&source_data) {
+            if account.base.is_native() {
+                return Err(TokenError::NativeNotSupported.into());
+            }
+            Self::validate_owner(
+                program_id,
+                &account.base.owner,
+                authority_info,
+                authority_info.data_len(),
+                account_info_iter.as_slice(),
+            )?;
+        } else if let Ok(mint) = StateWithExtensions::<Mint>::unpack(&source_data) {
+            if &mint.base.mint_authority.expect("No mint authority") != authority_info.key {
+                return Err(TokenError::AuthorityTypeNotSupported.into());
+            }
+            if let COption::Some(mint_authority) = mint.base.mint_authority {
+                Self::validate_owner(
+                    program_id,
+                    &mint_authority,
+                    authority_info,
+                    authority_info.data_len(),
+                    account_info_iter.as_slice(),
+                )?;
+            } else {
+                return Err(TokenError::AuthorityTypeNotSupported.into());
+            }
+        } else if source_data.len() == Multisig::LEN {
+            Self::validate_owner(
+                program_id,
+                source_info.key,
+                authority_info,
+                authority_info.data_len(),
+                account_info_iter.as_slice(),
+            )?;
+        }
```

Comment:
Out of an absolute abundance of caution can you add an `else` case that errors? ie
```suggestion
        } else {
            return Err(TokenError::InvalidState.into());
        }
```